keras_nlp/models/generative_task.py

```diff
@@ -81,7 +81,7 @@ def wrapped_generate_function(
             import jax
 
             @jax.jit
-            def compiled_generate_function(inputs, end_token_id, state):
+            def compiled_generate_function(inputs, stop_token_ids, state):
```

Comment:
Fairly technical jax question, but what happens if we pass lists of differing lengths here? Does jax automatically recompile? We could possibly pass `static_argnames` here, but not sure the right approach.

Reply:
Every element in the list will be treated as an individual Jax variable and will trigger recompilation. I think `static_argnames` is the right call here just checking that they work well with lists

Reply:
We will have to force this to be a tuple to guarantee that stop tokens are immutable